src/spec.ts

```diff
@@ -388,7 +395,9 @@ function normalizeRangedUnit(spec: UnitSpec) {
 
 // FIXME(#1804): re-design this
 function normalizeOverlay(spec: UnitSpec, overlayWithPoint: boolean, overlayWithLine: boolean, config: Config): LayerSpec {
-  const {mark, selection, encoding, ...outerSpec} = spec;
+  // _ is used to denote a dropped property of the unit spec
+  // which should not be carried over to the layer spec
+  const {mark, selection, projection: _p, encoding, ...outerSpec} = spec;
```

Comment:
Why should we drop projection for point overlay over lines? 

(Suppose we draw lines between different cities and we want to overlay points over each city?)